src/codeManager.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { BaseExplorer } from "./baseExplorer";
+import { Constants } from "./constants";
+import { DeviceItem } from "./Model/DeviceItem";
+import { TelemetryClient } from "./telemetryClient";
+import { Utility } from "./utility";
+
+export class CodeManager extends BaseExplorer {

```

Comment:
do we need to extends BaseExplorer? 
Seems the output method is not used in this class. 
Besides, is the ouput in BaseExplorer more like a utility?